### PR TITLE
Add Prolog LSP implementation to list of supported languages

### PR DESCRIPTION
**Pull request by jamesnvc**
Following up on #2156

### CODE REVIEW

**Comment by ericdallo** [comment]
@jamesnvc Just making it clear that if you want, you can migrate the `lsp-register-client` from your readme to here, with that users just need to install lsp-mode and when opening a prolog major mode everything should work :)
Take a look on an already existing client like [lsp-clojure](https://github.com/emacs-lsp/lsp-mode/blob/master/clients/lsp-clojure.el) for example, LMK you need more info/help.

**Comment by jamesnvc** [comment]
> @jamesnvc Just making it clear that if you want, you can migrate the `lsp-register-client` from your readme to here, with that users just need to install lsp-mode and when opening a prolog major mode everything should work :)

Oh cool, will do, thank you!

**Comment by leungbk** [comment]
Thanks for your PR! Please make sure to update `lsp-language-id-configuration`.

**Review by ericdallo — APPROVED** [review]

File: clients/lsp-prolog.el [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                                       "--" "stdio")
+  "The prolog-lsp server command."
+  :group 'lsp-prolog
+  :risky t
+  :type 'list)
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection
+  (lsp-stdio-connection lsp-prolog-server-command)
```
Comment (kiennq) [inline clients/lsp-prolog.el:48]: You should make this a lambda that returns `lsp-prolog-server-command`. So user can configure it **after** `lsp-prolog` is loaded.

**Comment by jamesnvc** [comment]
> @kiennq commented on this pull request.
> You should make this a lambda that returns `lsp-prolog-server-command`. So user can configure it **after** `lsp-prolog` is loaded.

Ah, good point! Thank-you

**Comment by jcs090218** [comment]
New language? Cool! Thanks for the PR! 😄

**Review by yyoncho — APPROVED** [review]

**Pull request merged**